test/distributed/_composable/test_composability/test_3d_composability.py

```diff
@@ -0,0 +1,757 @@
+
+
+class Test3DComposability(FSDPTest):
+    logger = logging.getLogger()
+    logger.info("Starting job: 3D composability test")
+    DeviceType = Union[int, str, torch.device]
+
+    def _init_logger(self):
+        self.logger.setLevel(logging.INFO)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.INFO)
+        formatter = logging.Formatter(
+            "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+        )
+        ch.setFormatter(formatter)
+        self.logger.addHandler(ch)
+
+        os.environ["KINETO_LOG_LEVEL"] = "5"
+
+    @skip_if_lt_x_gpu(4)
+    def test_3d_composability(self):
+        self.run_subtests(
+            {
+                "dp_degree": [
+                    1,
+                ],
+                "tp_degree": [
+                    2,
+                ],
+                "pp_degree": [
+                    2,
+                ],
+                "float8": [True],
+                "async_tp": [True],
+                "schedule_class": ["1f1b"], #["1f1b","gpipe","interleaved_1f1b","flexible_interleaved_1f1b",]
+            },
+            self._test_3d_composability,
+        )
+        self.logger.info("Finished job: 3D composability test")
+
+    def _test_3d_composability(self, dp_degree, tp_degree, pp_degree, float8, async_tp, schedule_class):
+        TRACE_BUFFER_SIZE = "TORCH_NCCL_TRACE_BUFFER_SIZE"
+        TRACE_FILE = "TORCH_NCCL_DEBUG_INFO_TEMP_FILE"
+        DUMP_ON_TIMEOUT = "TORCH_NCCL_DUMP_ON_TIMEOUT"
+        ASYNC_ERROR_HANDLING = "TORCH_NCCL_ASYNC_ERROR_HANDLING"
+        SKIP_CLEANUP = "3"
+
+        world_size = int(dp_degree * tp_degree * pp_degree)
+        parallel_dims = ParallelDims(
+            dp=dp_degree,
+            tp=tp_degree,
+            pp=pp_degree,
+            world_size=world_size,
+            enable_loss_parallel=False,
+            dp_type="fsdp",
+            float8=float8,
+            async_tp=async_tp,
+            schedule_class=schedule_class,
+        )
+        device = torch.device("cuda", index=dist.get_rank())
+        torch.cuda.set_device(device)
+
+        def _warn_overwrite_env(env, val):
+            if env in os.environ:
+                self.logger.warning("ENV will be overridden based on job config")
+            os.environ[env] = val
+
+        _warn_overwrite_env(ASYNC_ERROR_HANDLING, SKIP_CLEANUP)
+        _warn_overwrite_env(TRACE_BUFFER_SIZE, str(0))
+
+        os.environ["TORCH_NCCL_AVOID_RECORD_STREAMS"] = "1"
+        torch.manual_seed(0)
+        model_args = ModelArgs(dropout_p=0.0)
+        model = Transformer(model_args)
+        world_mesh = parallel_dims.build_mesh(device_type="cuda")
+
+        if parallel_dims.float8:
+            scale_for_fsdp = False
+            if parallel_dims.dp_enabled and parallel_dims.dp_type == "fsdp":
+                scale_for_fsdp = True
+            # a no-op hander if float8 is not enabled
+            float8_handler = Float8Handler(parallel_dims, scale_for_fsdp)
```

Comment:
is float8 path tested in CI? it requires dependency on torchao

Reply:
It's tested right now, with torchao imported when `float8=True`

Reply:
I mean pytorch CI needs torchao as depdency. I believe local devgpu works because you installed torchao by yourself